### ISSUE TITLE
name:tag variables usage from side builder configurations

### ISSUE THREAD

**Issue by gaassassins** [hadolint/hadolint#516]
- [x] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Pass image name/tag check

### Actual behavior
The Dockerfiles in this project depend on that gradlew build had already run (and the jar files already exist). Completely working Dockerfile validation output anyway:
```bash
/dev/stdin:1:6 expecting the image name with a tag
```

### Steps to reproduce the behavior

Dockerfile:
```Dockerfile
FROM @dockerRegistry/my/test/company/image/java11:@buildVersion
ENV JAVA_OPTS="test"
```

build.gradle file:
```gradle
subprojects {
    tasks.replace("copy", Copy).configure {
        copy {
            from("src") {
                include 'Dockerfile'

                filter { it.replaceAll('@dockerRegistry', dockerRegistry) }
                filter { it.replaceAll('@buildVersion', project.version) }
            }
            into("build")
        }
    }
}

```

Command:
`docker run --rm -i hadolint/hadolint < Dockerfile`

Output:
```bash
/dev/stdin:1:6 expecting the image name with a tag
```
**Output of 
  `docker run --rm hadolint/hadolint hadolint --version`:** 
Haskell Dockerfile Linter v1.19.0-46-g06208fc

**Additional environment details (OS, stack version, etc.)**
Running under our CI (docker). Local runs with base OS Linux, but dockerized.

**Comment by m-ildefons** [comment]
I'm not sure this is fixable on the Hadolint level, because if I understood you correctly, you are using gradle to text-replace something in the Dockerfile.
Would it be a  possibility for you to run Hadolint after gradle?

**Comment by lorenzo** [comment]
agreed, we cannot validate files with a custom templating in them. We can only validate once the template has replaced the variables.
